File: domain/src/main/java/com/ortin/gesturetranslator/domain/usecases/WordCompileUseCase.java
package com.ortin.gesturetranslator.domain.usecases;

import com.ortin.gesturetranslator.domain.repository.WordCompilerRepository;

public class WordCompileUseCase {

    private final WordCompilerRepository wordCompilerRepository;

    public WordCompileUseCase(WordCompilerRepository wordCompilerRepository) {
        this.wordCompilerRepository = wordCompilerRepository;
    }

    public void addLetter(String letter) {
        wordCompilerRepository.addLetter(letter);
    }

    public void clearState() {
        wordCompilerRepository.clearState();
    }

    public String getWord() {
        return wordCompilerRepository.getWord();
    }
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/usecases/WordCompileUseCase.kt
package com.ortin.gesturetranslator.domain.usecases

import com.ortin.gesturetranslator.domain.repository.WordCompilerRepository
import javax.inject.Inject

class WordCompileUseCase @Inject constructor(private val wordCompilerRepository: WordCompilerRepository) {
    fun addLetter(letter: String) {
        wordCompilerRepository.addLetter(letter)
    }

    fun clearState() {
        wordCompilerRepository.clearState()
    }

    fun getWord(): String {
        return wordCompilerRepository.getWord()
    }
}
